tools/topology/topology2/cavs/cavs-nocodec.conf

```diff
@@ -120,9 +122,25 @@ Object.Dai {
 # Pipeline ID:1 PCM ID: 0
 Object.Pipeline {
 	# playback pipelines
-	passthrough-be.1 {
-		index	2
-		direction	playback
+	host-copier-gain-mixin-playback.1 {
+		index	1
+
+		Object.Widget.pipeline.1 {
+			stream_name 'NoCodec-0'
+		}
+		Object.Widget.copier.1 {
+			stream_name 'SSP0 Playback'
+		}
+		Object.Widget.gain.1 {
+			Object.Control.mixer.1 {
+				name 'Playback Volume 0'
+			}
+		}
+	}
+
+	mixout-gain-dai-copier-playback.1 {
+		index 2
+
```

Comment:
@RanderWang please follow the example for topology1/cavs-nocodec.conf

-> ALL the paths are modified from passthrough to mixin-mixout
-> only SSP0 will have a second PCM mixed to limit resources needed and reduce validation time. However for tests, we will be able to quickly add other mixers on all paths without having to do another major rewrite of the file.

Makes sense?

Reply:
@plbossart  Sure. please check this picture
![no](https://user-images.githubusercontent.com/39945651/191911160-6a6c3808-37e2-4493-8c30-3f54d9f5115a.png)

Reply:
According to topology1/cavs-nocodec.m4,  I don't add mixin & mixout to capture

Reply:
sounds good for playback @RanderWang 

For capture I am less clear on how we would introduce multiple streams. @bardliao had a topology with three copiers, here with the gain7.1, gain9.1 and gain11.1 I wonder what the next steps would be.

Reply:
@bardliao Can you add multiple capture based on my PR ?

Reply:
> @bardliao Can you add multiple capture based on my PR ?

Sure, I will do it.

Reply:
> sounds good for playback @RanderWang
> 
> For capture I am less clear on how we would introduce multiple streams. @bardliao had a topology with three copiers, here with the gain7.1, gain9.1 and gain11.1 I wonder what the next steps would be.

@plbossart  The topology that I show you is to add an additional host to DMIC DAI. I will create a PR for it. My plan is to add gain in the DMIC pipeline and create an identical pipeline connect to the DMIC DAI. What do you think?

Reply:
> > sounds good for playback @RanderWang
> > For capture I am less clear on how we would introduce multiple streams. @bardliao had a topology with three copiers, here with the gain7.1, gain9.1 and gain11.1 I wonder what the next steps would be.
> 
> @plbossart The topology that I show you is to add an additional host to DMIC DAI. I will create a PR for it. My plan is to add gain in the DMIC pipeline and create an identical pipeline connect to the DMIC DAI. What do you think?

so the difficulty is that we need to have 'endpoint' processing, i.e. a 'master' gain, before we fork to two paths which have each their own 'stream' processing, i.e. a 'local' gain.

see what I did for topology1 with the SSP0 capture path. 
![sof-adl-nocodec](https://user-images.githubusercontent.com/1149035/191927721-ef833291-6db4-4242-a6e6-f25f8de42bc5.png)

I wonder if this means we need to have 4 copiers (1 DAI, one after master gain, two PCM copiers)? There's no equivalent to DEMUX in IPC4, so I am *guessing* the DEMUX is replaced by a copier but that's just me thinking with my poor understanding of IPC4.

Reply:
actually for the DMIC path the 'master' gain would be replaued by the EQIIR

Reply:
As I know we need Gain for PCH DMIC to adjust volume @singalsu  ?

Reply:
Sorry, I missed this. A key usage for volume comp for dmic is mic mute. Some notebooks have the Mic mute button that should control it. The DMIC DAI does not expose any mute though it could be done in HW too. I guess now the UCM sets the PGAs to 0 dB by default. We can with IIR set the mic path sensitivity to be roughly OK. The PGA with sufficient adjust range is good for the user or some gain set wizard app to set up suitable sensitivity.

Reply:
Good points @RanderWang and @singalsu, for the DMIC topologies we do need the EQIIR to boost the volume and deal with more DC rejection, we need a user configuration gain and a switch to mute.

All PGAs should have both gain and switch btw, but we routinely forget the latter even in existing IPC3/topology1 cases.